RandBLAS/random_gen.hh

```diff
@@ -9,12 +9,28 @@
 #include <math.h>
 #include <Random123/array.h>
 #include <Random123/philox.h>
-#include <Random123/uniform.hpp>
-#include <Random123/philox.h>
 #include <Random123/threefry.h>
+#include <Random123/aes.h>
+#include <Random123/ars.h>
 #include <Random123/boxmuller.hpp>
+#include <Random123/uniform.hpp>
+
+/// @cond
+/// extend r123::float2 to work with structured bindings
+namespace std {
+  template<> struct tuple_size<r123::float2> { static constexpr size_t value = 2; };
+  template<> struct tuple_element<0, r123::float2> { using type = float; };
+  template<> struct tuple_element<1, r123::float2> { using type = float; };
+}
```

Comment:
Is modifying the std namespace like this considered common practice?

Reply:
yes, from what I've read this is the only way.

Reply:
here's a simple write up https://blog.tartanllama.xyz/structured-bindings/

Reply:
update: I'll remove these since `r123::flkoat2` works out of the box with structured bindings due to its all public non-static member variables.

Reply:
resolved in b5bdf79a